cryptofeed/exchanges/coinbase.py

```diff
@@ -13,27 +15,49 @@
 
 from yapic import json
 
+from cryptofeed.config import Config
 from cryptofeed.connection import AsyncConnection, RestEndpoint, Routes, WebsocketEndpoint
-from cryptofeed.defines import BID, ASK, BUY, COINBASE, L2_BOOK, L3_BOOK, SELL, TICKER, TRADES
+from cryptofeed.defines import BID, ASK, BUY, COINBASE, L2_BOOK, SELL, TRADES
 from cryptofeed.feed import Feed
 from cryptofeed.symbols import Symbol
 from cryptofeed.exchanges.mixins.coinbase_rest import CoinbaseRestMixin
-from cryptofeed.types import OrderBook, Ticker, Trade
-
+from cryptofeed.types import OrderBook, Trade
 
 LOG = logging.getLogger('feedhandler')
 
 
+def get_private_parameters(config: Config, chan: str = None, product_ids_str: list = None,
+                           rest_api: bool = False, endpoint: str = None) -> dict:
+    timestamp = str(int(time.time()))
+    if rest_api:
+        base_endpoint = '/api/v3/brokerage/'
+        endpoint = base_endpoint + endpoint
+        message = f'{timestamp}GET{endpoint}'
+    else:
+        product_ids_str = ",".join(product_ids_str)
+        message = f"{timestamp}{chan}{product_ids_str}"
+    signature = hmac.new(
+        config["coinbase"]["key_secret"].encode("utf-8"),
+        message.encode("utf-8"),
+        digestmod=hashlib.sha256,
+    ).hexdigest()
+    if rest_api:
+        return {'CB-ACCESS-KEY': config["coinbase"]["key_id"], 'CB-ACCESS-TIMESTAMP': timestamp,
```

Comment:
instead of self.config["coinbase"].... you can just use self.key_id, self.key_secret, etc

Reply:
This is not a class method, it's a stand alone function, so 'self' is not available. I decided not to integrate it in the class as it's also called by the 'symbols' class method (which overwrites the parent method from the Exchange object). We need to do that as authentication is needed even to retrieve symbols on Coinbase

Reply:
yeah i see that now